x/evm/handler.go

```diff
@@ -104,11 +106,23 @@ func handleMsgEthereumTx(ctx sdk.Context, k *Keeper, msg types.MsgEthereumTx) (*
 
 	config, found := k.GetChainConfig(ctx)
 	if !found {
+		if !st.Simulate {
+			refundErr := st.RefundGas(ctx)
+			if refundErr != nil {
+				panic(refundErr)
+			}
```

Comment:
Too many repeated code segment, it is recommended to extract it as one function.

Reply:
@louisliu2048 Done.